pkg/controller/kogitobuild/build/image_stream_kogito.go

```diff
@@ -0,0 +1,280 @@
+		},
+	}
+
+	//tagDefaultAnnotations lists the default annotations for ImageStreamTags
+	tagDefaultAnnotations = map[string]map[string]string{
+		KogitoQuarkusUbi8Image: {
+			"iconClass":   "icon-jbpm",
+			"description": "Runtime image for Kogito based on Quarkus native image",
+			"tags":        "runtime,kogito,quarkus",
+			"supports":    "quarkus",
+		},
+		KogitoQuarkusJVMUbi8Image: {
+			"iconClass":   "icon-jbpm",
+			"description": "Runtime image for Kogito based on Quarkus JVM image",
+			"tags":        "runtime,kogito,quarkus,jvm",
+			"supports":    "quarkus",
+		},
+		KogitoQuarkusUbi8s2iImage: {
+			"iconClass":   "icon-jbpm",
+			"description": "Platform for building Kogito based on Quarkus",
+			"tags":        "builder,kogito,quarkus",
+			"supports":    "quarkus",
+		},
+		KogitoSpringBootUbi8Image: {
+			"iconClass":   "icon-jbpm",
+			"description": "Runtime image for Kogito based on SpringBoot",
+			"tags":        "runtime,kogito,springboot",
+			"supports":    "springboot",
+		},
+		KogitoSpringBootUbi8s2iImage: {
+			"iconClass":   "icon-jbpm",
+			"description": "Platform for building Kogito based on SpringBoot",
+			"tags":        "builder,kogito,springboot",
+			"supports":    "springboot",
+		},
+	}
+)
+
+// resolveKogitoImageNameTag resolves the ImageStreamTag to be used in the given build, e.g. kogito-quarkus-ubi8-s2i:0.11
+func resolveKogitoImageNameTag(build *v1alpha1.KogitoBuild, isBuilder bool) string {
+	return strings.Join([]string{
+		resolveKogitoImageName(build, isBuilder),
+		resolveKogitoImageTag(build, isBuilder),
+	}, ":")
+}
+
+// resolveKogitoImageTag resolves the ImageTag to be used in the given build, e.g. 0.11
+func resolveKogitoImageTag(build *v1alpha1.KogitoBuild, isBuilder bool) string {
+	image := &build.Spec.RuntimeImage
+	if isBuilder {
+		image = &build.Spec.BuildImage
+	}
+	if len(image.Tag) > 0 {
+		return image.Tag
+	}
+	return infrastructure.GetKogitoImageVersion()
+}
+
+// resolveKogitoImageName resolves the ImageName to be used in the given build, e.g. kogito-quarkus-ubi8-s2i
+func resolveKogitoImageName(build *v1alpha1.KogitoBuild, isBuilder bool) string {
+	image := &build.Spec.RuntimeImage
+	if isBuilder {
+		image = &build.Spec.BuildImage
+	}
+	if len(image.Name) > 0 {
+		return image.Name
+	}
+	imageName := kogitoImages[build.Spec.Runtime][isBuilder]
+	if build.Spec.Native && !isBuilder {
+		imageName = KogitoQuarkusUbi8Image
+	}
+	return imageName
+}
+
+// resolveKogitoImageName resolves the ImageName to be used in the given build, e.g. kogito-quarkus-ubi8-s2i
+func resolveKogitoImageStreamName(build *v1alpha1.KogitoBuild, isBuilder bool) string {
+	imageName := resolveKogitoImageName(build, isBuilder)
+	image := &build.Spec.RuntimeImage
+	if isBuilder {
+		image = &build.Spec.BuildImage
+	}
+	if len(image.Name) > 0 { // custom image
+		return strings.Join([]string{customKogitoImagePrefix, imageName}, "")
```

Comment:
what is the purpose to add custom prefix here?:

Reply:
See the example @sutaakar gave: when an user creates their own s2i images, we will create custom ImageStreams to handle those instead of using our own. We need a new name in case the user sets their image name being the same of ours.